cmd/clusterctl/pkg/client/cluster/inventory.go

```diff
@@ -105,7 +115,8 @@ func (p *inventoryClient) EnsureCustomResourceDefinitions() error {
 	}
 
 	// Install the CRDs.
-	for _, o := range objs {
+	for i := range objs {
+		o := objs[i]
```

Comment:
Sidenote, but i'm actually starting to wonder if this linter is actually useful. Reusing the range pointer is only bad when doing concurrent work which we're not here, or any other part of the codebase.

cc @ncdc

Reply:
Reusing the range pointer is bad if the address is stored in another data structure with the expectation that it's unique each time through the loop, as well as when passing it to a function (e.g. for concurrency). I do agree that if we have a call like `client.Delete(ctx, obj)` and it's the range pointer, that's ok, but the linter does not know that. I'd rather be deliberately safe and defensive in our code than put in `//nolint` "because we know better" - one day things might change downstream and our nolint would introduce unwanted behavior.

Reply:
Yeah, that sounds good to me. The main issue here is that I think readability is taking a hit for a small gain. Like you said, most of our uses are actually safe.